runtime/lua/vim/mouse.lua

```diff
@@ -0,0 +1,56 @@
+--- Event handling for `<MouseMove>`, allowing for multiple subscribers
+---
+--- Example:
+---
+--- ```lua
+--- vim.mouse.subscribe(function(pos)
+---  print('Mouse moved to ' .. pos.screenrow .. ', ' .. pos.screencol)
+--- end)
+--- ```
+local M = {}
+
+---@type fun(mouse: MouseInfo)[]
+local subscribers = {}
+
+local function handle()
+  ---@class MouseInfo
+  ---@field screenrow number Screen row
+  ---@field screencol number Screen column
+  ---@field winid number Window the cursor is over
+  ---@field winrow number Row inside "winid"
+  ---@field wincol number Column inside "winid"
+  ---@field line number Line inside "winid"
+  ---@field column number Column inside "winid"
+  local mouse = vim.fn.getmousepos()
+
+  for _, fn in ipairs(subscribers) do
+    fn(mouse)
+  end
+end
+
+---Subscribe the provided handler to the `<MouseMove>` event
+---
+---Handlers receive a table, equivalent to the result
+---of `vim.fn.getmousepos()`:
+---
+---  - screenrow (number)
+---  - screencol (number)
+---  - winid (number)
+---  - winrow (number)
+---  - wincol (number)
+---  - line (number)
+---  - column (number)
+---@param handler fun(mouse: MouseInfo)
+function M.subscribe(handler)
+  table.insert(subscribers, handler)
+end
+
+function M.setup()
+  if vim.version().minor > 8 and vim.o.mousemoveevent then
```

Comment:
checking only `minor` is the wrong way to check nvim version. could use `has('nvim-0.8')` instead.

Reply:
We shouldn't check the version here in any case, as we _know_ what version we are running under (0.10.0 or higher).

Reply:
Definitely was one of the things I wasn't sure about. And my bad, should've marked this as a draft.